mlflow/xgboost.py

```diff
@@ -255,19 +256,48 @@ def log_model(
 
 
 def _load_model(path):
-    import xgboost as xgb
+    """
+    Load Model Implementation.
+
+    :param path: Local filesystem path to
+                    the MLflow Model with the ``xgboost`` flavor (MLflow < x.x.x) or
+                    the top-level MLflow Model directory (MLflow >= x.x.x).
+    """
+    import importlib
 
-    model = xgb.Booster()
-    model.load_model(os.path.abspath(path))
+    model_dir = os.path.dirname(path) if os.path.isfile(path) else path
+    flavor_conf = _get_flavor_configuration(model_path=model_dir, flavor_name=FLAVOR_NAME)
+
+    if "data" in flavor_conf:
+        # XGBoost Booster models saved in MLflow (<x.x.x) specify
+        # the ``data`` field within its flavor configuration.
+        # In this case, we create a Booster() instance and load model weights.
+        model_class = "xgboost.core.Booster"
+        xgb_model_path = os.path.join(model_dir, flavor_conf["data"])
+    else:
+        # In contrast, XGBoost models saved in new MLflow (>=x.x.x) do not
+        # specify the ``data`` field within its flavor configuration.
+        # We use ``model_class`` to specify its XGBoost model class.
+        # In this case, we first get the XGBoost model from
+        # its flavor configuration and then create an instance based on its class.
+        model_class = flavor_conf.get("model_class", "xgboost.core.Booster")
+        xgb_model_path = os.path.join(model_dir, "model.xgb")
```

Comment:
Is it possible to keep the `data` field and remove if-else here? 

```python
model_class = flavor_conf.get("model_class", "xgboost.core.Booster")
xgb_model_path = os.path.join(model_dir, flavor_conf["data"])
module, cls = model_class.rsplit(".", maxsplit=1)
model = getattr(importlib.import_module(module), cls)()
model.load_model(xgb_model_path)
```

Reply:
> ```
> xgb_model_path = os.path.join(model_dir, flavor_conf["data"])
> ...
> model.load_model(xgb_model_path)
> ```
Thanks for your suggestion! I think we can do:
> xgb_model_path = os.path.join(model_dir, flavor_conf.get("data", "model.xgb"))

When `model_class` is saved, `data` is not a key of `flavor_conf` anymore. So `flavor_conf["data"]` returns None. In that case, we can manually append the saved model file to `xgb_model_path`.

Reply:
I kept the comments in the latest commit. They are meant to explain the difference in old / new model saving / loading. But if they look redundant, we can remove them.

Reply:
> When `model_class` is saved, data is not a key of flavor_conf anymore.

Can we save both `model_class` and `data`?

Reply:
> > When `model_class` is saved, data is not a key of flavor_conf anymore.
> 
> Can we save both `model_class` and `data`?

I think this is also fine. But in `pyfunc.load_model()`, `DATA` is already added a model path if it exists in MLmodel conf ([L666](https://github.com/mlflow/mlflow/blob/master/mlflow/pyfunc/__init__.py#L666)). It adds additional work to [retrieving model directory](https://github.com/mlflow/mlflow/pull/4954/files#diff-f6bd607049d4969297151a812f447bdc13453a856df017983f7ea81470d25bd0R268) every time `_load_pyfunc()` is called.

Reply:
Or we can save `data` in xgboost flavor but not in pyfunc flavor to avoid adding `DATA` path in `pyfunc.load_model()`. Please let me know which is preferred. Thanks!

Reply:
It came to me that we don't need to save `model_class` in pyfunc flavor as well, since it is not necessary when calling `pyfunc.load_model`, and loading is handled by local `_load_pyfunc`.